bin/linter.dart

```diff
@@ -185,7 +185,8 @@ Future runLinter(List<String> args, LinterOptions initialLintOptions) async {
         machineOutput: options['machine'],
         quiet: options['quiet'])
       ..write();
-  } catch (err, stack) {
+    // ignore: avoid_catching_errors
+  } on Error catch (err, stack) {
```

Comment:
I think it is actually appropriate to catch anything that might be thrown in a top-level method like this. I'd remove the `on` clause and ignore the lint.